dashboard/app/dsl/script_dsl.rb

```diff
@@ -82,19 +83,30 @@ def stage(name, properties = {})
     if @stage
       @stages << {
         stage: @stage,
+        visible_after: @stage_visible_after,
         scriptlevels: @scriptlevels,
         stage_extras_disabled: @stage_extras_disabled,
       }.compact
     end
     @stage = name
     @stage_flex_category = properties[:flex_category]
     @stage_lockable = properties[:lockable]
+    @stage_visible_after = determine_visible_after_time(properties[:visible_after])
     @scriptlevels = []
     @concepts = []
     @skin = nil
     @stage_extras_disabled = nil
   end
 
+  def determine_visible_after_time(visible_after_value)
+    current_time = Time.now
+    raw_diff_to_wed = 3 - current_time.wday
+    diff_to_wed = raw_diff_to_wed > 0 ? raw_diff_to_wed : 7 + raw_diff_to_wed
```

Comment:
nit: I think you can just use `raw_diff_to_wed % 7` here, because it will return a non-negative number even when the input is negative:
```
irb(main):001:0> 10 % 7
=> 3
irb(main):002:0> 3 % 7
=> 3
irb(main):003:0> -4 % 7
=> 3
```